Project_0.Altis/AI/Unit/ActionUnitArrest.sqf

```diff
@@ -28,144 +29,177 @@ CLASS("ActionUnitArrest", "Action")
 			T_SETV("stateChanged", true);
 			T_SETV("stateMachine", 2);
 		};
+
+		if (IS_TARGET_ARRESTED_UNCONSCIOUS_DEAD) then {
+			OOP_INFO_0("ActionUnitArrest: completed, reason: target unit dead, unconscious or arrested."); 
+			T_SETV("stateChanged", true);
+			T_SETV("stateMachine", 3);
+		};
 		
-		pr _state = ACTION_STATE_ACTIVE;
 		scopename "switch";
 		switch (T_GETV("stateMachine")) do {
 
-			/*
-			--------------------------------------------------------------------------------------------------------------------------------------------
-			|	 C A T C H  U P  																									   				   |
-			--------------------------------------------------------------------------------------------------------------------------------------------
-			*/
+			// CATCH UP
 			case 0: {
-				OOP_INFO_0("ActionUnitArrest: Chasing target.");
-				
+				OOP_DEBUG_1("CATCH UP %1", getPos _captor distance2D getPos _target);
+
+				if (
+					getPos _captor distance2D getPos _target < 6 && 
+					!(IS_TARGET_ARRESTED_UNCONSCIOUS_DEAD) &&
+					random 4 <= 2
+				) then {
+					[[_target], {
+						params ["_target"];
+						if (!hasInterface) exitWith {};
+
+						OOP_DEBUG_0("uncon");
+						_target playMoveNow "unconsciousfacedown"; // face plant
+						sleep 2;
+						_target playMoveNow "acts_aidlpsitmstpssurwnondnon01"; // sitting down and tied up
+
+					}] remoteExec ["spawn", _target, false];
+
+					_target setVariable ["timeArrested", time+10];
+
+					REMOTE_EXEC_CALL_STATIC_METHOD("UndercoverMonitor", "onUnitArrested", [_target], _target, false);	
+					
+					T_SETV("stateMachine", 1);
+					breakTo "switch";
+				};
+
+				if (IS_TARGET_ARRESTED_UNCONSCIOUS_DEAD) exitWith {
+					T_SETV("state", ACTION_STATE_COMPLETED);
+					ACTION_STATE_COMPLETED
+				};
+
 				if (T_GETV("stateChanged")) then {
 					T_SETV("stateChanged", false);
-					T_SETV("stateTimer", time);		
+					T_SETV("stateTimer", time);
 					
 					_captor dotarget _target;
-					
+
 					pr _handle = [_target,_captor] spawn {
-						params["_target","_captor"];
-						waitUntil{
-							_pos = (eyeDirection _target vectorMultiply 1.6) vectorAdd getpos _target;
+						params ["_target", "_captor"];
+						waitUntil {
+							pr _pos = (eyeDirection _target vectorMultiply 1.6) vectorAdd getpos _target;
+							
 							_captor doMove _pos;
 							_captor doWatch _target;
 							_pos_arrest = getpos _target;
-							sleep 0.5;
-							_isMoving = !(_pos_arrest distance getpos _target <0.1);
+
+							if (getpos _target distance getpos _captor > 30) then { sleep 3;};
+							sleep 1;
+
+							_isMoving = !(_pos_arrest distance getpos _target < 0.1);
 							_target setVariable ["isMoving", _isMoving];
 							
-							_return = !_isMoving && {_pos distance getpos _captor < 1.5};
+							pr _return = !_isMoving && {_pos distance getpos _captor < 1.5};
 							_return
 						};
 					};
+					
 					T_SETV("spawnHandle", _handle);
```

Comment:
Please, before you spawn it here and at another place, make sure there is no previous instance of "spawnHandle". If it's not a scriptNull, then terminate it.

It might be required If we for whatever reason will decide to re-enter this or some other state later.

We don't want to have scheduler full of scripts.

Reply:
Hum this is marvis code but as i understand it, he handle this with the stateChanged variable.
[HERE](https://github.com/Sparker95/Project_0/blob/add-warning-shots-to-arrest-goal/Project_0.Altis/AI/Unit/ActionUnitArrest.sqf#L122) and [HERE](https://github.com/Sparker95/Project_0/blob/add-warning-shots-to-arrest-goal/Project_0.Altis/AI/Unit/ActionUnitArrest.sqf#L177)

If you think it's not ok i'll take a look